doc/scripts/genrest/genrest.py

```diff
@@ -96,15 +63,148 @@ def print_items(items, outdir, indent):
+
+    if len(sys.argv) != 3:
+        print("usage: {} <Kconfig> <output directory>", file=sys.stderr)
+        sys.exit(1)
+
+    kconf = kconfiglib.Kconfig(sys.argv[1])
+    out_dir = sys.argv[2]
+
+    with open(os.path.join(out_dir, "index.rst"), "w") as index_rst:
+        index_rst.write(INDEX_RST_HEADER)
+
+        # - Sort the symbols by name so that they end up in sorted order in
+        #   index.rst
+        #
+        # - Use set() to get rid of duplicates for symbols defined in multiple
+        #   locations.
+        for sym in sorted(set(kconf.defined_syms), key=lambda sym: sym.name):
+            # Write an RST file for the symbol
+            write_sym_rst(sym, out_dir)
+
+            # Add an index entry for the symbol that links to its RST file.
+            # Also list its prompt(s), if any. (A symbol can have multiple
+            # prompts if it has multiple definitions.)
+            #
+            # The strip() avoids RST choking on stuff like *foo *, when people
+            # accidentally include leading/trailing whitespace in prompts.
+            index_rst.write("   * - :option:`CONFIG_{}`\n     - {}\n"
+                            .format(sym.name,
+                                    " / ".join(node.prompt[0].strip()
+                                               for node in sym.nodes
+                                               if node.prompt)))
+
+def write_sym_rst(sym, out_dir):
+    # Writes documentation for 'sym' to <out_dir>/CONFIG_<sym.name>.rst
+
+    kconf = sym.kconfig
+
+    with open(os.path.join(out_dir, "CONFIG_{}.rst".format(sym.name)),
+              "w") as sym_rst:
+
+        # List all prompts on separate lines
+        prompt_str = "\n\n".join("*{}*".format(node.prompt[0].strip())
+                                 for node in sym.nodes if node.prompt) \
+                     or "*(No prompt -- not directly user assignable.)*"
+
+        # - :orphan: suppresses warnings for the symbol RST files not being
+        #   included in any toctree
+        #
+        # - '.. title::' sets the title of the document (e.g. <title>). This
+        #   seems to be poorly documented at the moment.
+        sym_rst.write(":orphan:\n\n"
+                      ".. title:: {0}\n\n"
+                      ".. option:: CONFIG_{0}\n\n"
+                      "{1}\n\n"
+                      "Type: ``{2}``\n\n"
+                      .format(sym.name,
+                              prompt_str,
+                              kconfiglib.TYPE_TO_STR[sym.type]))
+
+        # Symbols with multiple definitions can have multiple help texts
+        for node in sym.nodes:
+            if node.help is not None:
+                # Wrap the help text in a code block to preserve "ASCII art"
+                # and the like
+                sym_rst.write("Help\n"
+                              "====\n\n"
+                              ".. code-block:: none\n\n"
```

Comment:
I think there are folks using markup tags in the help text for lists and such that won't render as expected if treated as a code block.

Reply:
@dbkinder 
Not sure about Zephyr, but I've seen quite a lot of help texts with figures drawn in text in the Linux kernel.

I agree that the output is uglier compared to a plain paragraph.

Reply:
Oh, you mean they use RST formatting in help texts? Yeah, in that case, it's definitely wrong to wrap it in a `code-block`.

Reply:
Talked a bit to @mbolivar on IRC. It wouldn't take anything fancy for the formatting to get messed up either. Take a look at `SYS_LOG_SPI_LEVEL` in `drivers/spi/Kconfig` for example.

Looks like you can preserve newlines in RST by putting `|` at the beginning of each line. There are probably other cases that would get messed up still though. I think a `code-block` would be safest.

Reply:
I think the help text should be presented as text like it used to be, using code-block here looks really bad. The few cases were this get messed up we can live with or fix in Kconfig.